### PR TITLE
Update topbar with newer styles

### PR DESCRIPTION
**Pull request by avatus**
Some changes have been requested to be added to the new topbar. Mostly, a dedicated "resources" button tab, and some icon changes. The updated figma is found here https://www.figma.com/file/JKd0Bgs30AnNfNCLUuzzc2/Unified-Resources?type=design&node-id=0-1&mode=design&t=wMhjcNdUYsHMFaeb-0

Screenshot 2024-01-12 at 3 31 48 PM

### CODE REVIEW

**Review by ibeckermayer — APPROVED** [review]
Nit: These sub-labels are redundant when the icon is already labeled:
image

image

Consider updating the storybook to allow forcing the nav into various states for visual inspection.

File: web/packages/teleport/src/TopBar/TopBar.tsx [hunk 1]
```diff
@@ -191,29 +173,37 @@ export function TopBar({ CustomLogo }: TopBarProps) {
...
         {!hasDockedElement && assistEnabled && (
           <ButtonIconContainer onClick={() => setShowAssist(true)}>
             <BrainIcon />
           </ButtonIconContainer>
         )}
         <Notifications />
         <UserMenuNav username={ctx.storeUser.state.username} />
+        {showAssist && (
+          <Suspense fallback={null}>
+            <Assist onClose={() => setShowAssist(false)} />
```
Comment (kimlisa) [inline web/packages/teleport/src/TopBar/TopBar.tsx:194]: oops, did we test out if assist looks ok? (i forgot about it)

File: web/packages/teleport/src/TopBar/TopBar.tsx [hunk 2]
```diff
@@ -191,29 +173,37 @@ export function TopBar({ CustomLogo }: TopBarProps) {
...
-          </>
-        )}
+          </Flex>
+        </>
+      )}
+      {feature?.hideFromNavigation && (
+        <ButtonIconContainer onClick={handleBack}>
+          <ArrowLeft size="medium" />
+        </ButtonIconContainer>
+      )}
```
Comment (kimlisa) [inline web/packages/teleport/src/TopBar/TopBar.tsx:183]: what does this do?

**Review by kimlisa — COMMENTED** [review]
some of our alignment issues came back:

image

image

File: web/packages/teleport/src/TopBar/TopBar.tsx (hunk 2 above)
Comment (avatus) [reply web/packages/teleport/src/TopBar/TopBar.tsx:183]: This is shown when using TAG. We remove all the navigation buttons and replace it with this (previously existing) arrow back component. this keeps the current functionality of Access Graph

File: web/packages/teleport/src/TopBar/TopBar.tsx (hunk 1 above)
Comment (avatus) [reply web/packages/teleport/src/TopBar/TopBar.tsx:194]: Pic of docked assist (the rest of the assit view modes overlay ontop and dont affect the width)
Screenshot 2024-01-12 at 5 09 36 PM

Comment (avatus) [reply web/packages/teleport/src/TopBar/TopBar.tsx:201]: and the overlay  (cant really see much cuz its just on top but just showing the zindex wrking correctly)
Screenshot 2024-01-12 at 5 10 24 PM

Comment (kimlisa) [reply web/packages/teleport/src/TopBar/TopBar.tsx:201]: thanks for the check 👍

**Review by kimlisa — APPROVED** [review]

**Pull request merged**